packages/jsii/lib/assembler.ts

```diff
@@ -391,18 +391,20 @@ export class Assembler implements Emitter {
                 this._visitDocumentation(constructor, jsiiType.initializer);
             }
         } else if (jsiiType.base) {
-            const baseType = this._dereference(jsiiType.base);
-            if (!baseType) {
-                this._diagnostic(type.symbol.valueDeclaration,
-                                 ts.DiagnosticCategory.Error,
-                                 `Unable to resolve type ${jsiiType.base.fqn} (base type of ${jsiiType.fqn})`);
-            } else if (spec.isClassType(baseType)) {
-                jsiiType.initializer = baseType.initializer;
-            } else {
-                this._diagnostic(type.symbol.valueDeclaration,
-                    ts.DiagnosticCategory.Error,
-                    `Base type of ${jsiiType.fqn} (${jsiiType.base.fqn}) is not a class`);
-            }
+            this._defer(() => {
+                const baseType = this._dereference(jsiiType.base!);
+                if (!baseType) {
+                    this._diagnostic(type.symbol.valueDeclaration,
+                                    ts.DiagnosticCategory.Error,
+                                    `Unable to resolve type ${jsiiType.base!.fqn} (base type of ${jsiiType.fqn})`);
+                } else if (spec.isClassType(baseType)) {
+                    jsiiType.initializer = baseType.initializer;
```

Comment:
I am concerned about `defer`. For example, the fact that you only assign the initializer here looks like a potential race condition. How do I know what order these `defer` statements actually execution? Now `initializer` is a moving target!

I am okay with using `defer` for validation (maybe should be renamed to `validate`), but it seems very fragile for mutations of the model